pkg/kubectl/kubectl.go

```diff
@@ -34,13 +34,73 @@ type kubeCtlVersionData struct {
 	ServerVersion *kubeCtlVersionInfo `json:"serverVersion,omitempty"`
 }
 
+// Example stdout:
+// Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", ..., Platform:"linux/amd64"}
+// Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", ..., Platform:"linux/amd64"}
+
+// We don't care about the exact reason why the parsing failed, we'll display
+// more context in the error message anyway.
+var errParsing = errors.New("parse error")
+
+func parseVersionLine(line string, versionInfo **kubeCtlVersionInfo) error {
+	// Only interested in what's between '{', '}'
+	idx := strings.Index(line, "{")
+	list := line[idx+1 : len(line)-2]
+
+	parts := strings.Split(list, ",")
+	for _, part := range parts {
+		// parts are of the form key:"value"
+		part := strings.TrimSpace(part)
+		colon := strings.Index(part, ":")
+		if colon == -1 {
+			return errParsing
+		}
+		key := part[0:colon]
+
+		if key == "GitVersion" {
+			value := part[colon+2 : len(part)-1]
+			*versionInfo = &kubeCtlVersionInfo{GitVersion: value}
+		}
+	}
+
+	return nil
```

Comment:
I would prefer if this function returns the version info, rather then setting it in place. IMHO it's hard to see at first whats happening and also this way strange mutations can happen in the function and it's hard to debug. WDYT?

Reply:
Souds like a good idea, changed it.

Reply:
Did you forget to push?

Reply:
Nope, that's the last commit in the PR (and still all green!)